File: domain/src/main/java/com/beat/domain/promotion/exception/PromotionErrorCode.java
package com.beat.domain.promotion.exception;

import com.beat.global.common.exception.base.BaseErrorCode;

public enum PromotionErrorCode implements BaseErrorCode {
	/*
	404 NotFound
	*/
	PROMOTION_NOT_FOUND(404, "해당 홍보 정보를 찾을 수 없습니다.");
	private final int status;
	private final String message;

	PromotionErrorCode(int status, String message) {
		this.status = status;
		this.message = message;
	}

	public int getStatus() {
		return status;
	}

	public String getMessage() {
		return message;
	}
}

File: domain/src/main/kotlin/com/beat/domain/promotion/exception/PromotionErrorCode.kt
package com.beat.domain.promotion.exception

import com.beat.domain.exception.DomainErrorCode
import com.beat.domain.exception.DomainErrorType

enum class PromotionErrorCode(
    override val code: String,
    override val type: DomainErrorType,
    override val message: String,
) : DomainErrorCode {
    TOO_MANY_CAROUSEL_PROMOTIONS(
        "PROMOTION_TOO_MANY_CAROUSEL_PROMOTIONS",
        DomainErrorType.INVALID_INPUT,
        "노출 가능한 캐러셀 프로모션 수를 초과했습니다.",
    ),
    ;
}
